File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int j = n-1;
        int i = 0;
        while(j>=0 and nums[j]==val){
            j--;
        }
        while(i<j){
            if(nums[i]==val){
                swap(nums[i], nums[j]);    
            }
            while(j>=0 and nums[j]==val){
                j--;
            }
            i++;
        }
        return j+1;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    private void swap(int[] nums, int a, int b){
        int temp = nums[a];
        nums[a] = nums[b];
        nums[b] = temp;
    }
    public int removeDuplicates(int[] nums) {
        int i = 0; // denote unique element
        int n = nums.length;
        int j = 0;
        while(j<n){
            while(j<n && nums[j]==nums[i]){
                j++;
            }
            if(j==n) return i+1;

            int k = j;
            while(k != i+1){
                swap(nums, k, k-1);
                k--;
            }
            i++;
            j++;
        }
        return i+1;
    }
}